src/router/handlers/user.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * @swagger
+ * /user/:
+ *  get:
+ *    description: Use to request a user
+ *    parameters:
+ *      - name: ha-api-token
+ *        in: header
+ *        required: true
+ *        schema:
+ *          type: string
+ *          format: uuid
+ *    responses:
+ *      '200':
+ *        description: Success
+ *        schema:
+ *          type: object
+ *          properties:
+ *            user:
+ *              type: object
+ */
+let getUser = async (req, res) => {
+    
+};
+
+/**
+ * @swagger
+ * /user/:
+ *  put:
+ *    description: Use to update a user
+ *    parameters:
+ *      - name: ha-api-token
+ *        in: header
+ *        required: true
+ *        schema:
+ *          type: string
+ *          format: uuid
+ *    responses:
+ *      '200':
+ *        description: Success
+ *      '400':
+ *        description: 'Invalid syntax'
+ *      '401':
+ *        description: 'JWT not found in header'
+ *      '403':
+ *        description: 'JWT does not have user privileges'
+ *      '404':
+ *        description: 'Not found'
+ */
+let updateUser = async (req, res) => {
+    
+};
+
+/**
+ * @swagger
+ * /user/:
+ *  delete:
+ *    description: Use to delete a user
+ *    parameters:
+ *      - name: ha-api-token
+ *        in: header
+ *        required: true
+ *        schema:
+ *          type: string
+ *          format: uuid
+ *    responses:
+ *      '200':
+ *        description: Success
+ *      '400':
+ *        description: 'Invalid syntax'
+ *      '401':
+ *        description: 'JWT not found in header'
+ *      '403':
+ *        description: 'JWT does not have user privileges'
+ *      '404':
+ *        description: 'Not found'
+ */
+let deleteUser = async (req, res) => {
+    
+};
+
+/**
+ * @swagger
+ * /hacks/:hid/reg/users/csv/:
```

Comment:
In terms of file structure, you've divided up routes as those that are `open` and those that are `admin`, which makes some sense. Following that model, this route should be an `admin` route. However, now we're using a separate way of dividing things up, based on function: `user` vs `everything else`. I kind of like to see all the users grouped together by function like it is in `user.js`, but I think consistency is a good idea. Up to you!

Reply:
>  I kind of like to see all the users grouped together by function like it is in user.js

I am not totally sure what you mean by this. Could you elaborate on it for me?

Reply:
Ha okay, I think I was mixed up a bit—terminology is tricky! In one of your comments, you said you were trying to reorganize based on permissions: certain endpoints are `open` (so no authentication required), and others are only for `admin` (authentication required). I think now I understand: `user` is supposed to be endpoints for general users who are logged in without admin privileges. Sorry for the confusion!

If that is the case, getting the user registration form (the csv) should be an `admin` route (general users should not be able to access it)!